libraries/tools/kotlin-stdlib-gen/src/templates/Generators.kt

```diff
@@ -594,11 +594,7 @@ object Generators : TemplateGroupBase() {
             val first = ArrayList<T>()
             val second = ArrayList<T>()
             for (element in this) {
-                if (predicate(element)) {
-                    first.add(element)
-                } else {
-                    second.add(element)
-                }
+                (if (predicate(element)) first else second).add(element)
```

Comment:
There's a similar pattern in line 620 for CharSequences and Strings, would you want to improve it as well.